crypto/src/elgamal.rs

```diff
@@ -0,0 +1,383 @@
+use crate::errors::CryptoError;
+use crate::types::{Ciphertext, PrivateKey};
+
+use num_bigint::BigUint;
+
+#[cfg(feature = "tracing")]
+use tracing::{debug, error};
```

Comment:
Most likely, you don't need this feature and this conditional compilation. The `tracing` crate is lightweight and will not cause a huge overhead.

Reply:
Got it, thanks!

Will still keep it as a reference for the future myself